src/Tribe/Views/V2/Widgets/Compatibility.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Compatibility for Advanced List Widgets and List Widget.
+ * Scenarios:
+ * * TEC with Pro Activated in V1 ( Reverse )
+ * * Pro Disabled to TEC in V1 ( Default ) - @todo PRO widget removed, free widgets stay
+ * * V1 to V2 ( Default )
+ * * Pro Disabled V2 Active ( Default )
+ * * Pro Active V2 to V1 with constant ( Reverse )
+ * * Pro Disabled V2 back to V1 with constant ( Default ) - @todo PRO widget removed, free widgets stay
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Pro\Views\V2\Widgets
+ */
+
+namespace Tribe\Events\Views\V2\Widgets;
+
+/**
+ * Class Compatibility
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Widgets
+ */
+class Compatibility {
+
+	/**
+	 * The default primary list widget id base string.
+	 *
+	 * @var string
+	 */
+	protected $primary_id_base = 'tribe-events-list-widget';
+
+	/**
+	 * The default alternative list widget id base string.
+	 *
+	 * @var string
+	 */
+	protected $alternative_id_base = 'tribe-events-adv-list-widget';
+
+	/**
+	 * Adds the filters for V2 Widgets
+	 *
+	 * @since TBD
+	 */
+	public function hooks() {
+		add_action( 'tribe_plugins_loaded', [ $this, 'switch_compatibility' ] );
+		add_filter( 'option_sidebars_widgets', [ $this, 'remap_list_widget_id_bases' ] );
+	}
+
+	/**
+	 * Switches the primary and alternative id base when v1 is active
+	 * or v2 widgets are disabled, this enables support for upgrading from
+	 * v1 free list widget to the v1 Pro advanced list widget.
+	 *
+	 * @since TBD
+	 */
+	public function switch_compatibility() {
+		// if Pro is disabled, use the defaults.
+		if ( ! class_exists( 'Tribe__Events__Pro__Main' ) ) {
+			return;
+		}
+
+		if (
+			! tribe_events_views_v2_is_enabled() ||
+			tribe_events_widgets_v2_is_disabled()
+		) {
+			$this->primary_id_base     = 'tribe-events-adv-list-widget';
+			$this->alternative_id_base = 'tribe-events-list-widget';
+		}
+
+		add_filter( "option_widget_{$this->primary_id_base}", [ $this, 'merge_list_widget_options' ] );
+	}
+
+	/**
+	 * Remap the widget id_base for the Pro Advanced List Widget.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $widget_areas An array of widgets areas with the saved widgets in each location.
+	 *
+	 * @return array<string,mixed> $widget_areas An array of widgets areas with the saved widgets in each location.
+	 */
+	public function remap_list_widget_id_bases( $widget_areas ) {
+
```

Comment:
```suggestion
```

Here and elsewhere, remove blank lines at start and end of function and methods definitions.